plugins/mimecast/komand_mimecast/util/api.py

```diff
@@ -0,0 +1,182 @@
+        sig = base64.encodebytes(hmac_sha1).rstrip()
+        sig = sig.decode("UTF-8")
+
+        # Create request headers
+        return {
+            "Authorization": f"MC {self.access_key}:{sig}",
+            "x-mc-app-id": self.app_id,
+            "x-mc-date": hdr_date,
+            "x-mc-req-id": request_id,
+            "Content-Type": "application/json",
+        }
+
+    def _handle_error_response(self, response: dict):
+        for errors in response.get("fail", []):
+            for error in errors.get("errors", []):
+                if error.get(CODE) == XDK_BINDING_EXPIRED_ERROR:
+                    raise PluginException(
+                        cause=ERROR_CASES.get(XDK_BINDING_EXPIRED_ERROR),
+                        assistance="Please provide a valid AccessKey.",
+                        data=response,
+                    )
+                elif error.get(CODE) in ERROR_CASES:
+                    raise PluginException(
+                        cause=ERROR_CASES.get(error.get(CODE)),
+                        assistance=BASIC_ASSISTANCE_MESSAGE,
+                        data=response,
+                    )
+                elif error.get(CODE) == FIELD_VALIDATION_ERROR:
+                    raise PluginException(
+                        cause=f"This {error.get('field')} field is mandatory; it cannot be NULL.",
+                        assistance=BASIC_ASSISTANCE_MESSAGE,
+                        data=response,
+                    )
+                elif error.get(CODE) == VALIDATION_BLANK_ERROR:
+                    raise PluginException(
+                        cause=f"This {error.get('field')} field, if present, cannot be blank or empty.",
+                        assistance=BASIC_ASSISTANCE_MESSAGE,
+                        data=response,
+                    )
+                else:
+                    self.logger.error(response)
+                    raise PluginException(preset=PluginException.Preset.UNKNOWN, data=response)
+
+    def _handle_rest_call(
+        self,
+        method: str,
+        uri: str,
+        data: dict = None,
+        meta_data: dict = None,
+    ) -> dict:
+
+        payload = {DATA_FIELD: ([data] if data is not None else [])}
+
+        if meta_data is not None:
+            payload[META_FIELD] = meta_data
+
+        try:
+            request = requests.request(
+                method=method.upper(), url=f"{self.url}{uri}", headers=self._prepare_header(uri), data=str(payload)
+            )
+        except requests.exceptions.RequestException as e:
+            raise PluginException(preset=PluginException.Preset.SERVER_ERROR, data=e)
+
+        try:
+            response = request.json()
+        except json.decoder.JSONDecodeError:
+            raise PluginException(
+                cause="Unknown error.",
+                assistance="The Mimecast server did not respond correctly. Response not in JSON format. Response in logs.",
+            )
```

Comment:
```suggestion
        try:
            response = request.json()
        except json.decoder.JSONDecodeError as e:
            raise PluginException(
                cause="Unknown error.",
                assistance="The Mimecast server did not respond correctly. Response not in JSON format. Response in logs.", data=e
            )
```
or data=response.text